lib/plugins.py

```diff
@@ -545,6 +572,15 @@ def __init__(self, parent, config, name):
                 hooks[aname].append((self, func))
                 self._hooks_i_registered.append((aname,func))
 
+        # collect names of all class attributes with ._is_daemon_command
+        self._daemon_commands = tuple(attrname for attrname in dir(type(self))
```

Comment:
Hmm. Why save `self._daemon_commands` to an instance attribute? It seems to not be accessed outside this function.. couldn't a temp var just work fine here?

If you plan on adding commits though, that make use of this instance dict: may I suggest at least clearing it on `.close()` (see below)..?